src/librustc_passes/diagnostics.rs

```diff
@@ -248,4 +248,6 @@ register_diagnostics! {
     E0472, // asm! is unsupported on this target
     E0561, // patterns aren't allowed in function pointer types
     E0571, // `break` with a value in a non-`loop`-loop
+    E0583, // `break` with no label in the condition of a `while` loop
+    E0584, // `continue` with no label in the condition of a `while` loop
```

Comment:
I was thinking that 

(a) maybe it makes sense to have just one condition code here ("`break` or `continue` with no label in the condition of a `while` loop")

and

(b) given that we do technically accept `while break { }` in some situations (only if in another loop, etc), it is probably a good idea to write a more extended explanation and in particular to show an example of how to add a label (i.e., `'a: while break 'a { }`)

What do you think?